docs/src/main/sphinx/functions/setdigest.rst

```diff
@@ -0,0 +1,164 @@
+=========================
+Set Digest functions
+=========================
+
+Trino offers several functions that deal with the
+`MinHash <https://en.wikipedia.org/wiki/MinHash>`_ technique.
+
+MinHash is used to quickly estimate the
+`Jaccard similarity coefficient <https://en.wikipedia.org/wiki/Jaccard_index>`_
+between two sets.
+
+It is commonly used in data mining to detect near-duplicate web pages at scale.
+By using this information, the search engines efficiently avoid showing
+within the search results two pages that are nearly identical.
+
+The following naive example showcases how the Set Digest functions are
+used to estimate the similarity between texts::
+
+    WITH text_input(id, text) AS (VALUES
+                                 (1, 'The quick brown fox jumps over the lazy dog'),
+                                 (2, 'The quick red fox jumps over the lazy dog'),
+                                 (3, 'The quick and the lazy'),
+                                 (4, 'The quick brown fox jumps over the dog')),
+         text_ngrams(id, ngrams) AS (
+             SELECT id,
+                    transform(
+                      ngrams(
+                        split(text, ' '),
+                        4
+                      ),
+                      token -> array_join(token, ' ')
+                    )
+             FROM text_input
+         ),
+         minhash_digest(id, digest) AS (
+             SELECT id,
+                    (SELECT make_set_digest(v) FROM unnest(ngrams) u(v))
+             FROM text_ngrams
+         ),
+         setdigest_side_by_side(id1, digest1, id2, digest2) AS (
+             SELECT m1.id as id1,
+                    m1.digest as digest1,
+                    m2.id as id2,
+                    m2.digest as digest2
+             FROM (SELECT id, digest FROM minhash_digest) m1
+             JOIN (SELECT id, digest FROM minhash_digest) m2
+               ON m1.id != m2.id AND m1.id < m2.id
+        )
+    SELECT id1,
+           id2,
+           intersection_cardinality(digest1, digest2) AS intersection_cardinality,
+           jaccard_index(digest1, digest2)            AS jaccard_index
+    FROM setdigest_side_by_side
+    ORDER BY id1, id2;
+
+which results in:
+
+.. code-block:: text
+
+     id1 | id2 | intersection_cardinality |    jaccard_index
+    -----+-----+--------------------------+---------------------
+       1 |   2 |                        3 | 0.16666666666666666
+       1 |   3 |                        0 |                 0.0
+       1 |   4 |                        4 |                 0.6
+       2 |   3 |                        0 |                 0.0
+       2 |   4 |                        1 |                 0.0
+       3 |   4 |                        0 |                 0.0
+
```

Comment:
Add some sentence that explains how to interpret the data .. along the lines of .. row one shows similarilry between sentences 1 and 2 are quite high (only difference being red vs brown fox), row two shows that 1 and 3 are very different ...

Reply:
Good point Manfred. Just throwing a lengthy SQL will probably just confuse a newbie reader. I've added an explanation of what the statement is doing and also a short interpretation of the results.